src/main/java/com/google/devtools/build/lib/skyframe/RepositoryMappingFunction.java

```diff
@@ -196,17 +211,23 @@ public SkyValue compute(SkyKey skyKey, Environment env)
    *     Optional.empty().
    */
   private Optional<RepositoryMappingValue> computeForBazelModuleRepo(
-      RepositoryName repositoryName, BazelDepGraphValue bazelDepGraphValue) {
+      RepositoryName repositoryName,
+      BazelDepGraphValue bazelDepGraphValue,
+      Set<RepositoryName> repositoryOverrides) {
     ModuleKey moduleKey = bazelDepGraphValue.getCanonicalRepoNameLookup().get(repositoryName);
     if (moduleKey == null) {
       return Optional.empty();
     }
+    RepositoryMapping repoMapping = bazelDepGraphValue.getFullRepoMapping(moduleKey);
+    if (repositoryName.isMain()) {
+      repoMapping =
+          repoMapping.withAdditionalMappings(
+              repositoryOverrides.stream().collect(toMap(RepositoryName::getName, name -> name)));
```

Comment:
(tests are failing because `REPOSITORY_OVERRIDES` would now need to be injected in more tests, I can do that once we have agreed on the functionality)

Reply:
could we maybe call the flag `--inject_repository`, similar to `inject_repo` vs `override_repo`? There's a nice bit of symmetry there, at least superficially... Then we could also say `--inject_repository=X=foo` always creates the repo with the canonical repo name of `+_injected+X`, or something.

Reply:
I completely rewrote this PR. It now implements `--inject_repository` through a simulated `use_repo_rule` call for a `local_repository`, which guarantees behavior identical to that of a regular module extension repo.